types/query.d.ts

```diff
@@ -579,7 +579,7 @@ declare module 'mongoose' {
     snapshot(val?: boolean): this;
 
     /** Sets the sort order. If an object is passed, values allowed are `asc`, `desc`, `ascending`, `descending`, `1`, and `-1`. */
-    sort(arg?: string | { [key: string]: SortOrder } | undefined | null): this;
+    sort(arg?: string | { [key: string]: SortOrder } | { score: { $meta: 'textScore' } } | undefined | null): this;
```

Comment:
The sort key doesn't necessarily have to be called `score`. You can also do `{ textScore: { $meta: 'textScore' } }`, see https://thecodebarbarian.com/2014/04/10/a-nodejs-perspective-on-whats-new-in-mongodb-2-6-part-i-text-search.html .

According to https://www.mongodb.com/docs/manual/reference/operator/aggregation/meta/, `{ $meta: 'indexKey' }` is also valid.

Maybe something like `{ [key: string]: { $meta: 'textScore' | 'indexKey' } }` ?

Reply:
Good point about the `score` key. `{ $meta: 'indexKey' }` should be valid but not for sorting.